packages/navigator/src/electron-browser/electron-navigator-menu-contribution.ts

```diff
@@ -54,6 +64,28 @@ export class ElectronNavigatorMenuContribution implements MenuContribution, Comm
             isEnabled: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
             isVisible: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
         }));
+        commands.registerCommand(OPEN_WITH_SYSTEM_APP, UriAwareCommandHandler.MonoSelect(this.selectionService, {
+            execute: async uri => {
+                this.openWithSystemApplication(uri);
+            },
+            isEnabled: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
+            isVisible: uri => !!this.workspaceService.getWorkspaceRootUri(uri),
+        }));
+        this.openWithService.registerHandler({
+            id: 'system-editor',
+            label: nls.localize('theia/navigator/systemEditor', 'System Editor'),
+            providerName: nls.localizeByDefault('Built-in'),
+            // Low priority to avoid conflicts with other open handlers.
+            canHandle: uri => (uri.scheme === 'file') ? 10 : 0,
+            open: uri => {
+                this.openWithSystemApplication(uri);
+                return {};
+            }
+        });
+    }
+
+    protected openWithSystemApplication(uri: URI): void {
+        window.electronTheiaCore.openWithSystemApp(uri['codeUri'].fsPath);
```

Comment:
Use `FileUri.fsPath` instead, please. It handles special cases.

Reply:
It does not make any difference in this case (electron will be the same OS between browser and back-end), but in general, we should not do any OS-specific file manipulation on the browser side.

Reply:
I would recommend `Uri#path#fsPath()` instead of `FileUri.fsPath`, since it deals better with the case where backend and frontend are running on different OS.

Reply:
We have ~100 usages of `FileUri.fsPath()`. If it's not the right thing, we have a problem. Should FileUri.fsPath forward to `Uri`?

Reply:
`FileUri.fsPath()` used to be exclusively used on the backend (I believe you moved it from `node` to `common` recently), where only available OS was the backend OS. Changing it to forward to `Uri#path#fsPath` would probably be a good idea though.

Reply:
Good, I will use `FileUri.fsPath()` and in other PR somebody need to redirect from `FileUri.fsPath()` to `Uri#path#fsPath`. Do we have an issue for that already?

Reply:
Nope

Reply:
Okay, I will create one.
Otherwise, all suggestions are applied, please have a look.

Reply:
Opened #13620